src/test/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumerTest.java

```diff
@@ -142,10 +163,85 @@ public void tearDown() throws IOException, InterruptedException {
         nullProject.delete();
         freeStyleProject.delete();
         workflowJob.delete();
+        events.clear();
+    }
+
+    @Test
+    public void testMirrorSynchronizedUpdateAndDeleteFiresBothEvents() throws InterruptedException {
+        BitbucketServerConfiguration serverConfiguration = mock(BitbucketServerConfiguration.class);
+        when(bitbucketPluginConfiguration.getServerById(bitbucketSCM.getServerId())).thenReturn(Optional.of(serverConfiguration));
+        when(serverConfiguration.getBaseUrl()).thenReturn(BITBUCKET_BASE_URL);
+        MirrorSynchronizedWebhookEvent refsChangedWebhookEvent = new MirrorSynchronizedWebhookEvent(
+                BITBUCKET_USER,
+                new BitbucketMirrorServer("1", "mirror1"),
+                MIRROR_SYNCHRONIZED.getEventId(),
+                new Date(),
+                refChanges(BitbucketRefChangeType.DELETE, BitbucketRefChangeType.UPDATE),
+                bitbucketRepository,
+                BitbucketRepositorySynchronizationType.INCREMENTAL);
+
+        consumer.process(refsChangedWebhookEvent);
+
+        verify(bitbucketTrigger)
+                .trigger(
+                        eq(BitbucketWebhookTriggerRequest.builder().actor(BITBUCKET_USER).build()));
+        verify(workflowTrigger)
+                .trigger(
+                        eq(BitbucketWebhookTriggerRequest.builder().actor(BITBUCKET_USER).build()));
+
+        SCMHeadEvent<RefsChangedWebhookEvent> event = events.poll(500L, TimeUnit.MILLISECONDS);
+        assertThat(event.getPayload(), equalTo(refsChangedWebhookEvent));
```

Comment:
I fear this may be flaky since it imposes a strict order of the events, and we really don't care for the order. May I propose you do either two `poll` right after each other and put the resulting events in a list and do `assertThat(list, containsInAnyOrder(..))` OR you do a `events.drainTo(list)` which is susceptible to flaking since it has no wait times in it. I'm guessing two `poll` will be the most stable option.

Having said that, I'm happy to merge this as it is and see how we go, it may not cause any issues, but if it does both of us know how to fix it with relative ease.

Reply:
Yeah I specified the order because we always fire the update first- but that's no guarantee it'll be added by Jenkins first. I think assert contains in any order is the way to go, even if the order is consistent, it's irrelevant to the functionality we actually care about.